pkg/kontainerdrivermetadata/data_getter.go

```diff
@@ -1,166 +1,30 @@
 package kontainerdrivermetadata
 
 import (
-	"fmt"
-
-	mVersion "github.com/mcuadros/go-version"
 	v3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
 	"github.com/rancher/rancher/pkg/namespace"
 	rketypes "github.com/rancher/rke/types"
-	"github.com/rancher/rke/util"
-	"github.com/sirupsen/logrus"
-	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 func GetRKESystemImages(k8sVersion string, sysImageLister v3.RkeK8sSystemImageLister, sysImages v3.RkeK8sSystemImageInterface) (rketypes.RKESystemImages, error) {
-	name := k8sVersion
-	sysImage, err := sysImageLister.Get(namespace.GlobalNamespace, name)
+	sysImage, err := sysImageLister.Get(namespace.GlobalNamespace, k8sVersion)
 	if err != nil {
-		if !errors.IsNotFound(err) {
-			return rketypes.RKESystemImages{}, err
-		}
-		sysImage, err = sysImages.GetNamespaced(namespace.GlobalNamespace, name, metav1.GetOptions{})
+		sysImage, err = sysImages.GetNamespaced(namespace.GlobalNamespace, k8sVersion, metav1.GetOptions{})
 		if err != nil {
 			return rketypes.RKESystemImages{}, err
 		}
 	}
-	return sysImage.SystemImages, err
+	return sysImage.SystemImages, nil
 }
 
 func GetRKEAddonTemplate(addonName string, addonLister v3.RkeAddonLister, addons v3.RkeAddonInterface) (string, error) {
```

Comment:
Is there any reason to keep `GetRKEAddonTemplate` or `GetRKESystemImages` around?

Reply:
nope can be removed.